internal/controllers/state/clusteroperator.go

```diff
@@ -81,26 +82,26 @@ func (com *clusterOperatorManager) Refresh(ctx context.Context, cond []configv1.
 		operatorv1helpers.SetOperandVersion(&co.Status.Versions, configv1.OperandVersion{Name: "operator", Version: releaseVersion})
 	}
 
-	if _, err := com.kubeClient.ClusterOperatorUpdateStatus(ctx, co, metav1.UpdateOptions{}); err != nil {
+	if err = com.kubeClient.StatusUpdate(ctx, co); err != nil {
 		return fmt.Errorf("could not update ClusterOperator: %w", err)
 	}
 
 	return nil
 }
 
 func (com *clusterOperatorManager) getOrCreate(ctx context.Context) (*configv1.ClusterOperator, error) {
-	co, err := com.kubeClient.ClusterOperatorGet(ctx, com.operatorName, metav1.GetOptions{})
-	if err != nil {
+	co := &configv1.ClusterOperator{}
+
+	if err := com.kubeClient.Get(ctx, types.NamespacedName{Name: com.operatorName}, co); err != nil {
```

Comment:
what do you think about leaving the API as is, and just changing the inner implementation? I think that the code will be a little bit clearer, no?

Reply:
We need to group those functions into new types to be able to mock the Kubernetes client. 
The name changes do make things clearer in my opinion, e.g. I think that `clusterOperatorManager.Refresh()` is more obvious than `SpecialResourcesStatus()`.
I understand that it can make the review more painful though. I can revert to the old names, but I would really change them in the long run, as they can be quite confusing. Or did I miss your point?

Reply:
The new names for clusterOepratorManager are great, i was talking about keeping the old API names for the clients package, and just changing the inner implementation to runtime-client. For example,  kubeClient.ClusterOperatorGet sounds more clear to me then kubeClient.Get (with namespaces name). WDYT?

Reply:
I removed that commit from the PR. Let us things related to the Kubernetes client as they are for now while we investigate if we can come up with a better `pkg/clients` package.
@yevgeny-shnaidman PTAL at the rest of the PR 🙂